### PR TITLE
Trial mar

### PR DESCRIPTION
**Pull request by mar-garcia**

### CODE REVIEW

File: EIC_manually.R [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  (myfiles <- myfiles[grep(paste0("MIX ", mixnum, "[A-Z]"), myfiles)])
+}
+myfiles_pos <- myfiles[grep("POS", myfiles)]
+myfiles_neg <- myfiles[grep("NEG", myfiles)]
+
+
+# Get the information regarding the standards that are in the samples:
+std_info <- read.table(paste0("data/", study, ".txt"),
+                       sep = "\t", header = TRUE, as.is = TRUE)
+std_info$name <- c(substring(std_info$name, 1, 33))
```
Comment (jorainer) [inline EIC_manually.R:28]: Why was that needed? did `$name` contain some strange names?

File: EIC_manually.R [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
+}
+write.csv(std_info, "x.csv", row.names = FALSE)
```
Comment (jorainer) [inline EIC_manually.R]: If this is to convert the `"mzneut"` from `character` to `numeric`, an easier thing would be to replace the `paste` above with `as.numeric` (i.e. `td_info$mzneut[i] = as.numeric(std_info$formula[i])`). You would then no longer need this `write.csv`.

File: EIC_superposed.Rmd [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+# Set up parallel processing using 3 cores
+if (.Platform$OS.type == "unix") {
+    register(bpstart(MulticoreParam(2)))
+} else {
+    register(bpstart(SnowParam(2)))
+}
+```
+
+# Data import
+In the file `XXXX_files.txt` there is the information regarding
```
Comment (jorainer) [inline EIC_superposed.Rmd:38]: Very minor thing: for better readability it would be nice if you added a new line after the title `# Data import` and the text - also between descriptive text and the R code blocks.

File: EIC_superposed.Rmd [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+std_info$name <- c(substring(std_info$name, 1, 33))
+std_info <- std_info[!is.na(std_info[, grep(polarity, colnames(std_info))]),]
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline EIC_superposed.Rmd]: like above, please replace `paste` with `as.numeric` to ensure that column `"mzneut"` contains numeric values. You can then remove the lines below exporting and importing to/from csv files.

File: EIC_superposed.Rmd [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  
+  chr = chromatogram(data, 
+                     mz = c(mzvalue - da,  mzvalue + da),
+                     aggregationFun = "max")
+  
+  cmpname <- gsub("L-", "", std_info$name[i])
+  cmpname <- tolower(gsub(" .*", "", cmpname))
+  
+  if(study == "internal_standards"){ 
+    filename <- paste0("plots/IS_", cmpname, "_",
```
Comment (jorainer) [inline EIC_superposed.Rmd]: Suggestion: to be consistent with all other analyses we have, it might be better to save the plots into a folder *images/EIC_superposed* (that's the *standard* way how I name folders for analyses: *images/<name of Rmd creating the plots>*. Open for discussion.

Comment (jorainer) [reply EIC_superposed.Rmd]: Eventually it might also be good to have a `dir.create("images/EIC_superposed", recursive = TRUE, showWarnings = FALSE)` at the beginning of the file to ensure that the directory is available.

File: exclusion_mz.R [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        )
+
+max.len = max(length(NEG), length(POS))
+
+NEG = c(NEG, rep(NA, max.len - length(NEG)))
+POS = c(POS, rep(NA, max.len - length(POS)))
+
+exclusion_mz <- data.frame(NEG, POS)
+
+write.table(exclusion_mz, "data/exclusion_mz.txt", sep="\t", row.names = FALSE)
```
Comment (jorainer) [inline exclusion_mz.R:67]: would be eventually interesting to add also a column `"notes"` that could contain the information from the comments (such as `"Formic acid"`, `"RT 203 (IS)"` etc.

File: get_rt.Rmd [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+                      sep = "\t", header = TRUE, as.is = TRUE)
+std_info$name <- c(substring(std_info$name, 1, 33))
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline get_rt.Rmd]: Also here, please use `as.numeric` instead of `paste` and remove the code below exporting/importing to/from csv file.

**Review by jorainer — CHANGES_REQUESTED** [review]
Nice @mar-garcia !

I have just some minor things that would be nice if you could address them.

File: EIC_manually.R (hunk 1 above)
Comment (mar-garcia) [reply EIC_manually.R:28]: Not, but sometimes are very long, like the ones of internal standards (e.g., "Succinic acid-2,2,3,3-d4" , "Creatinine (N-Methhyl_D3)").

Comment (jorainer) [reply EIC_manually.R:28]: OK. Fine with me to keep it like that.

File: EIC_manually.R (hunk 2 above)
Comment (mar-garcia) [reply EIC_manually.R]: Super!
Change applied.

File: exclusion_mz.R (hunk 6 above)
Comment (mar-garcia) [reply exclusion_mz.R:67]: I've changed the structure of the code.
Let me know what do you think.

File: internal_standards_evaluation.Rmd [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+```{r}
+std_info <- read.table("data/internal_standards.txt",
+                      sep = "\t", header = TRUE, as.is = TRUE)
+std_info <- std_info[,1:5]
+std_info <- std_info[!is.na(std_info[, grep(polarity, colnames(std_info))]),]
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline internal_standards_evaluation.Rmd]: Please also replace `paste` with `as.numeric` here

File: spectra_manually.R [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+
+# Get the information regarding the standards that are in the samples:
+std_info <- read.table(paste0("data/", study, ".txt"),
+                       sep = "\t", header = TRUE, as.is = TRUE)
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline spectra_manually.R]: Also here `paste` -> `as.numeric`

File: spectras_max_ions.Rmd [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+                       sep = "\t", header = TRUE, as.is = TRUE)
+std_info$name <- substring(std_info$name, 1, 33)
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline spectras_max_ions.Rmd]: Also here `paste` -> `as.numeric`

File: standard_dilution_dose_response.Rmd [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+std_info <- read.table("data/standards_dilution.txt",
+                      sep = "\t", header = TRUE, as.is = TRUE)
+std_info <- std_info[!is.na(std_info[, grep(polarity, colnames(std_info))]) &
+                       !(is.na(std_info$RT)) &
+                       std_info$mix == mixnum,]
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```
Comment (jorainer) [inline standard_dilution_dose_response.Rmd]: And here :)

**Review by jorainer — CHANGES_REQUESTED** [review]
Thanks @mar-garcia ! Excellent! 

Just some last changes from `paste` to `as.numeric` please.

**Review by jorainer — APPROVED** [review]
Excellent 🎉
